dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcContext.java

```diff
@@ -56,7 +56,27 @@ public class RpcContext {
     /**
      * use internal thread local to improve performance
      */
-    private static final InternalThreadLocal<RpcContextAttachment> SERVER_LOCAL = new InternalThreadLocal<RpcContextAttachment>() {
+    private static final RpcContext AGENT_SERVER_CONTEXT = new RpcContextAttachment() {
+        @Override
+        public RpcContextAttachment setObjectAttachment(String key, Object value) {
+            if (value == null) {
+                attachments.remove(key);
+            } else {
+                RpcContext.getServerResponseContext().setAttachment(key, value);
+                attachments.put(key, value);
+            }
+            return this;
+        }
+    };
```

Comment:
override all methods in RpcContextAttachment, and proxy to serverResponse / clientResponse